AdyenCard/Components/GiftCardComponent/GiftCardComponent.swift

```diff
@@ -263,38 +302,59 @@ public final class GiftCardComponent: PartialPaymentComponent,
         delegate.showConfirmation(for: component)
     }
 
-    private func requestOrder() -> Result<Void, Swift.Error> {
+    // MARK: - Partial Payment flow
+
+    private func startPartialPaymentFlow(paymentData: PaymentComponentData) -> Result<Void, Swift.Error> {
         if let order = order {
-            submit(order: order)
+            submit(order: order, paymentData: paymentData)
             return .success(())
         }
         guard let partialPaymentDelegate = partialPaymentDelegate else {
             AdyenAssertion.assertionFailure(message: Error.missingPartialPaymentDelegate.localizedDescription)
             return .failure(Error.missingPartialPaymentDelegate)
         }
         partialPaymentDelegate.requestOrder { [weak self] result in
-            self?.handle(orderResult: result)
+            self?.handle(orderResult: result, paymentData: paymentData)
         }
         return .success(())
     }
 
-    private func handle(orderResult: Result<PartialPaymentOrder, Swift.Error>) {
-        orderResult.handle(success: submit(order:), failure: { delegate?.didFail(with: $0, from: self) })
+    private func handle(orderResult: Result<PartialPaymentOrder, Swift.Error>,
+                        paymentData: PaymentComponentData) {
+        orderResult.handle(success: {
+            submit(order: $0, paymentData: paymentData)
+        }, failure: {
+            delegate?.didFail(with: $0, from: self)
+        })
     }
 
-    private func submit(order: PartialPaymentOrder) {
-        delegate?.didSubmit(createPaymentData(order: order), from: self)
+    // MARK: - Submit payment
+
+    private func submit(order: PartialPaymentOrder, paymentData: PaymentComponentData) {
+        self.delegate?.didSubmit(paymentData.replacingOrder(with: order), from: self)
     }
 
-    private func createPaymentData(order: PartialPaymentOrder? = nil) -> PaymentComponentData {
-        let details = GiftCardDetails(paymentMethod: giftCardPaymentMethod,
-                                      cardNumber: numberItem.value,
-                                      securityCode: securityCodeItem.value)
+    // MARK: - PaymentData creation
+
+    private func createPaymentData(order: PartialPaymentOrder?, cardPublicKey: String) -> Result<PaymentComponentData, Swift.Error> {
+        do {
+            let card = Card(number: numberItem.value, securityCode: securityCodeItem.value)
+            let encryptedCard = try CardEncryptor.encrypt(card: card, with: cardPublicKey)
```

Comment:
this is probably out of scope of this PR, but the interface of `CardEncryptor.encrypt` is rather unusual. it does throw, but then can return possibly invalid data (optional `number` and `securityCode`), that has to be checked and throws an error in case they are `nil`. shouldn't this be done inside the `CardEncryptor.encrypt`?

Reply:
yes, it is out of scope :-D